src/olympia/amo/utils.py

```diff
@@ -861,21 +861,37 @@ class SafeStorage(FileSystemStorage):
 
     DEFAULT_CHUNK_SIZE = 64 * 2**10  # 64kB
 
-    def __init__(self, *args, **kwargs):
-        """If `user_media` arg is provided, location will be lazily resolved under the
-        appropriate user_media_path. If `user_media` is falsey MEDIA_ROOT is used - see
-        user_media_path for the resolution logic."""
-        if 'user_media' in kwargs:
-            self._user_media = kwargs.pop('user_media')
+    root_setting = 'STORAGE_ROOT'
```

Comment:
btw, the reason we need to pass in the setting name and lazily resolve - i.e. can't just pass in a `location` kwarg with `path(settings.MEDIA_PATH, xxx)` or whatever, in it - is that it gets resolved too early for the testing setup where the paths are overridden to per-test temporary directories.